### PR TITLE
#690 custom anno 4

### PR DESCRIPTION
**Pull request by djencks**
Implements Peter's idea for an XMLAttribute meta-annotation.

### CODE REVIEW

File: biz.aQute.bndlib/src/aQute/bnd/annotation/xml/XMLAttribute.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+@Target({
+	ElementType.ANNOTATION_TYPE
+})
+public @interface XMLAttribute {
+
+	String namespace();
+
+	String prefix() default "";
+
+	String[] embedIn();
```
Comment (bjhargrave) [inline biz.aQute.bndlib/src/aQute/bnd/annotation/xml/XMLAttribute.java]: Could this default to "*"?

**Comment by bjhargrave** [comment]
I don't really have much expertise here, but I don't have any objections to this.

**Comment by djencks** [comment]
I improved the defaults processing per BJ's comment: not specifying the namespace now actually works and embedIn defaults to {"*"}

**Comment by djencks** [comment]
Changed how classes are represented in the Annotation backing map to be TypeRefs and added a test that now Class valued annotation elements are processed correctly.

**Pull request merged**